src/job_worker/Jobs.jl

```diff
@@ -322,32 +460,24 @@ function handle_job(
         "ydist" => "$(input.y_dist)"
     )
 
-    @debug "Ascertaining file name"
-    cache_path = _cache_location(config)
-
-    # Use only the criteria relevant to regional assessment
-    regional_assess_criteria = extract_criteria(qp, suitability_criteria())
-    # NOTE: This is where we could add additional hash params to invalidate
-    # cache
-    job_id = create_job_id(regional_assess_criteria)
-
-    assessed_fn = joinpath(cache_path, "$(job_id)_$(reg)_suitable.tiff")
-    @debug "File name: $(assessed_fn)"
+    assessed_fn = build_regional_assessment_file_path(;
+        query_params=qp, region=reg, reef_type=rtype, ext="tiff", config
+    )
+    @debug "COG File name: $(assessed_fn)"
 
     if !isfile(assessed_fn)
         @debug "File system cache was not hit for this task"
         @debug "Assessing region $(reg)"
         assessed = assess_region(reg_assess_data, reg, qp, rtype)
 
-        @debug "Writing to $(assessed_fn)"
-        _write_tiff(assessed_fn, assessed)
+        @debug "Writing COG to $(assessed_fn)"
+        _write_cog(assessed_fn, assessed, config)
     else
         @info "Cache hit - skipping regional assessment process!"
+        @debug "Pulling out raster from cache"
+        assessed = Raster(assessed_fn; missingval=0, lazy=true)
     end
 
-    @debug "Pulling out raster"
-    assessed = Raster(assessed_fn; missingval=0, lazy=true)
-
```

Comment:
@ConnectedSystems  realised this was doing extra work - hence removal - assessed is already defined in the case where we just computed it - needs to be in the cache case instead